oda_api/cli.py

```diff
@@ -66,20 +66,24 @@ def get(obj, instrument, product, argument):
 
 @cli.group("token")
 @click.option("-s", "--secret", default=None)
+@click.option('--allow-invalid', is_flag=True, default=False)
 @click.pass_obj
-def tokencli(obj, secret):
+def tokencli(obj, secret, allow_invalid):
     obj['secret_key'] = secret
+    obj['token'] = discover_token(allow_invalid=allow_invalid)
+    obj['decoded_token'] = decode_oda_token(obj['token'], secret_key=obj['secret_key'])
+    obj['allow_invalid'] = allow_invalid
+
+    if obj['token'] is None:
+        logger.warn("no token found!")
+        return
```

Comment:
Perhaps this should go a couple of lines above? before decoding I mean

Reply:
Well, `decode_oda_token` will raise if `token is None` so this is really useless. I remove! And also I add test for this.